fix: Adjusting docs on all repos permissions

yhakbar opened the pull request.

ZachGoldberg commented on docs/2.0/docs/pipelines/installation/viagithubapp.md:
```diff
@@ -35,7 +35,17 @@ As of 2024/09/10, the Gruntwork.io GitHub App requests the following permissions
+
+  For most day-to-day operations, you should be able to get away with only having the app installed on your `infrastructure-live-root` repository. The exceptions to this are the following:
+
+  1. You are using the app to vend new repositories during initial bootstrap. This is the case for any customer bootstrapping optional repositories during initial setup.
```
This bullet doesn't read very clear to me -- what does it mean to vend repositories during bootstrap?

ZachGoldberg reviewed (changes requested):
I think this is a step in the right direction, but for me the copy is still pretty confusing.  What is the actual recommendation and why?  What happens if I don't follow the recommendation?

ZachGoldberg commented on docs/2.0/docs/pipelines/installation/viagithubapp.md:
```diff
@@ -35,7 +35,17 @@ As of 2024/09/10, the Gruntwork.io GitHub App requests the following permissions
-  The level of granularity available to customers when configuring the GitHub App installation is to either install the app on a per-repository basis or on an entire organization. Our recommendation is to install the app on the organization as a whole, as this will allow Gruntwork.io to do all it's meant to do. You should use your best judgement when deciding on this, however. If you are unsure,  ...
+  The level of granularity available to customers when configuring the GitHub App installation is to either install the app on a per-repository basis or on an entire organization. Our recommendation is to install the app on the organization as a whole, as this will allow Gruntwork.io to do all it's meant to do. You should use your best judgement when deciding on this, however.
+
+  For most day-to-day operations, you should be able to get away with only having the app installed on your `infrastructure-live-root` repository. The exceptions to this are the following:
```
This seems to be contrary to the advice provided in the prior paragaph - which says we recommend allowing it for the whole org.

yhakbar commented:
@ZachGoldberg , I adjusted the recommendation to be granular and added more context for the consequences of doing so.

FYI, I think we're going to need to change these docs a lot in order to keep them accurate.
